units/patch_intent_test.go

```diff
@@ -1626,6 +1626,74 @@ tasks:
 	s.Equal("my-task", dbChildPatch.VariantsTasks[0].Tasks[0])
 }
 
+func (s *PatchIntentUnitsSuite) TestTriggerAliasWithDownstreamRevision() {
+	specificRevision := model.Version{
+		Id:         "childProj-some-version",
+		Identifier: "childProj",
+		Revision:   "abc",
+		Requester:  evergreen.RepotrackerVersionRequester,
+	}
+	s.Require().NoError(specificRevision.Insert())
+
+	specificVersionParserProject := &model.ParserProject{}
+	s.Require().NoError(util.UnmarshalYAMLWithFallback([]byte(`
+buildvariants:
+- name: my-build-variant
+  display_name: my-build-variant
+  run_on:
+    - some-distro
+  tasks:
+    - my-task
+tasks:
+- name: my-task`), specificVersionParserProject))
+	specificVersionParserProject.Id = specificRevision.Id
+	s.Require().NoError(specificVersionParserProject.Insert())
+
+	childPatchAlias := model.ProjectAlias{
+		ProjectID: "childProj",
+		Alias:     "childProj-patch-alias",
+		Task:      "my-task",
+		Variant:   "my-build-variant",
+	}
+	s.Require().NoError(childPatchAlias.Upsert())
+
+	p := &patch.Patch{
+		Id:      mgobson.NewObjectId(),
+		Project: s.project,
+		Author:  evergreen.GithubPatchUser,
+		Githash: s.hash,
+	}
+	s.NoError(p.Insert())
+
+	u := &user.DBUser{
+		Id: evergreen.ParentPatchUser,
+	}
+	s.NoError(u.Insert())
+
+	projectRef, err := model.FindBranchProjectRef(s.project)
+	s.NotNil(projectRef)
+	s.NoError(err)
+	s.Require().Len(projectRef.PatchTriggerAliases, 1)
+	projectRef.PatchTriggerAliases[0].DownstreamRevision = "abc"
+
+	s.Len(p.Triggers.ChildPatches, 0)
```

Comment:
Is this check relevant to the test? It seems like `p` hasn't been used so far, so I would expect that `p` would still have no child patches just from the assignment on L1660-1664.

Reply:
It can be removed